extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -219,6 +222,33 @@ export class SafeframeHostApi {
     return attributes;
   }
 
+  /**
+   * Returns the canonical URL of the page, if the publisher allows
+   * it to be passed.
+   * @return {string|undefined}
+   * @visibleForTesting
+   */
+  maybeGetCanonicalUrl() {
+    // Don't allow for referrer policy same-origin,
+    // as Safeframe will always be a different origin.
+    // Don't allow for no-referrer.
+    const metaReferrer = this.win_.document.querySelector(
+        "meta[name='referrer']");
+    if (metaReferrer && (
```

Comment:
Given you're checking against 3 values, let's do a switch:

const canonicalURL = ...
const metaReferrer = ...
if (!metaReferrer) return canonicalURL;
switch (metaReferrer.getAttribute('content')) {
    case 'same-origin':
    case 'no-referrer':
        return;
    case 'origin':
        return parseUrl(canonicalUrl).origin;
}
return canonicalUrl;